mulog-kinesis/src/com/brunobonacci/mulog/publishers/kinesis.clj

```diff
@@ -62,10 +63,17 @@
 
 ;https://docs.aws.amazon.com/cli/latest/reference/kinesis/put-records.html
 (defn kinesis-publisher
-  [{:keys [stream-name] :as config}]
+  [{:keys [stream-name max-items] :as config}]
   {:pre [stream-name]}
-  (KinesisPublisher.
-    (as-> config $
-          (merge DEFAULT-CONFIG $))
-    (rb/agent-buffer (:max-items config))
-    (or (:transform config) identity)))
\ No newline at end of file
+  (let [cfg (as-> config $
+               (merge DEFAULT-CONFIG $)
+               (update $ :max-items min KINESIS-MAX-RECORDS-NUMBER))]
+    (if (and
+          (not (nil? max-items))
+          (> max-items KINESIS-MAX-RECORDS-NUMBER))
+      (println (format "!!! Provided %d buffer size will be capped to maximum allowed 500" max-items)))
+    (KinesisPublisher.
+      cfg
+      (rb/agent-buffer (:max-items cfg))
```

Comment:
this agent buffer should be something like 10K so that if one or more `put-records` fail, not all events are lost and will be retried.